ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
The point of these principles is to protect - cheaply - against forces of change. 

I don't know whether it will, you don't know for certain it wont. 

One of the guiding principles of clean architecture IMO is that if you can protect yourself with a tiny modification to your code, you should.

The accumulation of lots of these little things can make the code very viscous